src/Xhgui/Searcher/Pdo.php

```diff
@@ -146,10 +146,28 @@ public function getAll($options = array())
     {
         $sort = $options['sort'];
         $direction = $options['direction'];
-        $page = $options['page'];
-        $perPage = $options['perPage'];
+        $page = (int)$options['page'];
+        if ($page < 1) {
+            $page = 1;
+        }
+        $perPage = (int)$options['perPage'];
+        $url = $options['conditions']['url'] ?? "";
 
-        $stmt = $this->pdo->query("
+        $stmt = $this->pdo->prepare("
+          SELECT COUNT(*) AS count
+          FROM {$this->table}
+          WHERE simple_url LIKE :url
+        ");
+        $stmt->execute(['url' => '%'.$url.'%']);
+        $totalRows = $stmt->fetch(PDO::FETCH_ASSOC)['count'];
```

Comment:
seems you want [fetchColumn](https://www.php.net/manual/en/pdostatement.fetchcolumn.php) here.